Night_Wisp bombs demo patch

Night-Wisp opened the pull request:
* Fix two minor errors in the Bombs demo.

Fix two minor errors in the Bombs demo.

Error 1: When enabling/disabling easy corner, the enable/disable text was mixed up. This resulted in being asked to confirm disabling easy corner, when it was being enabled, and vice versa.
Error 2: When choosing not to switch easy corner, the checkmark in the menu showing it's state still swapped.

Fix 1: Swap the text for enable/disable in the confirm easy mode toggle dialog box.
Fix 2: Call GetMenuBar()->Check(bombsID_EASYCORNER, m_easyCorner); (Check always returns void, and GetMenuBar is only called from the MenuBar, after it's been set up) to set the checkmark's state, when needed.

Added myself in a Modified By section

This fixes issue #22763

Night-Wisp committed:
Night_Wisp bombs demo patch (#1)

* Fix two minor errors in the Bombs demo.

Fix two minor errors in the Bombs demo.

Error 1: When enabling/disabling easy corner, the enable/disable text was mixed up. This resulted in being asked to confirm disabling easy corner, when it was being enabled, and vice versa.
Error 2: When choosing not to switch easy corner, the checkmark in the menu showing it's state still swapped.

Fix 1: Swap the text for enable/disable in the confirm easy mode toggle dialog box.
Fix 2: Call GetMenuBar()->Check(bombsID_EASYCORNER, m_easyCorner); (Check always returns void, and GetMenuBar is only called from the MenuBar, after it's been set up) to set the checkmark's state, when needed.

Added myself in a Modified By section

vadz commented on demos/bombs/bombs.cpp:
```diff
@@ -196,9 +197,11 @@ void BombsFrame::OnEasyCorner(wxCommandEvent& WXUNUSED(event))
+    if(ok!=wxYES)return GetMenuBar()->Check(bombsID_EASYCORNER, m_easyCorner);
 
     m_easyCorner = !m_easyCorner;
+    
+    GetMenuBar()->Check(bombsID_EASYCORNER, m_easyCorner);
```
I don't think this is necessary, the check menu item should have been already toggled, no?

Night-Wisp replied:
Yes, it's been removed now.

vadz commented on demos/bombs/bombs.cpp:
```diff
@@ -196,9 +197,11 @@ void BombsFrame::OnEasyCorner(wxCommandEvent& WXUNUSED(event))
              );
 
-    if(ok!=wxYES)return;
+    if(ok!=wxYES)return GetMenuBar()->Check(bombsID_EASYCORNER, m_easyCorner);
```
Minor, but we don't put compound statements on the same line. Also, a comment would probably be useful:
```suggestion
    if(ok!=wxYES)
    {
        // Undo the change to the menu bar, to keep it in sync with the actual option value.
        return GetMenuBar()->Check(bombsID_EASYCORNER, m_easyCorner);
    }
```

vadz commented on demos/bombs/bombs.cpp:
```diff
@@ -3,6 +3,7 @@
 // Purpose:     Bombs game
 // Author:      P. Foggia 1996
 // Modified by: Wlodzimierz Skiba (ABX) since 2003
+// Modified by: Night_Wisp in 2022
```
We don't really use those any longer since the switch to Git, you will be credited as the commit author anyhow.

vadz reviewed (commented):
Thanks for the fixes, they seem correct to me, but I think there are a couple of minor improvements that could be made to make the PR even better.

vadz commented:
Oh, and there is also [this small issue](https://github.com/wxWidgets/wxWidgets/runs/8105455208?check_suite_focus=true)

Night-Wisp committed:
Fix the issues mentioned in the merge review.

Removed trailing whitespace from line 203.
Removed Modified By line.
Converted an if statement to multiple lines, instead of 1.
Removed an unnecessary statement setting the checkmark state, when it already toggled correctly.

Night-Wisp committed:
Force checks to restart.

Force the spell checker test to restart, since it was cancelled by an internal error.

Night-Wisp commented:
@vadz I fixed the prior issues. But two different checks failed, without any changes made to any other files. Something is probably causing an intermittent issue.

vadz commented:
Thanks for the fixes, I'll merge this soon.

The tests failures are unrelated, unfortunately there is some flakiness in `wxWebView` tests. I hope to be able to debug it one day, but for now we just have to live with spurious failures from time to time, sorry about this.
